src/ape_ethereum/provider.py

```diff
@@ -624,11 +624,34 @@ def get_receipt(
         )
         hex_hash = HexBytes(txn_hash)
 
+        txn = {}
+        if transaction := kwargs.get("transaction"):
+            # perf: If called `send_transaction()`, we should already have the data!
+            txn = (
+                transaction
+                if isinstance(transaction, dict)
+                else transaction.model_dump(by_alias=True, mode="json")
+            )
+
+        private = kwargs.get("private")
+
         try:
             receipt_data = dict(
                 self.web3.eth.wait_for_transaction_receipt(hex_hash, timeout=timeout)
             )
         except TimeExhausted as err:
+            # We don't auto-wait for acceptance or confirmations for private transactions.
```

Comment:
since we are going with set-everything-to-zero plan, im not sure this comment still holds true